File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        from collections import defaultdict
        ct = defaultdict(int)
        for a in arr:
            ct[a]+=1
        
        unique = defaultdict(bool)
        for a in ct:
            if unique[ct[a]] == False:
                unique[ct[a]] = True
            else:
                return False
        return True

File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        Map<Integer, Integer> m = new HashMap<Integer, Integer>();
        
        for(int i=0; i<arr.length; i++){
            if(m.containsKey(arr[i])){
                m.put(arr[i], m.get(arr[i]) + 1);
            }else{
                m.put(arr[i], 0);
            }
        }
        
        Set<Integer> s = new HashSet<Integer>();
        for(Integer key: m.keySet()){
            if(s.contains(m.get(key))){
                return false;
            }else{
                s.add(m.get(key));
            }
        }
        return true;
    }
}